lib/datadog/core/telemetry/emitter.rb

```diff
@@ -22,12 +22,18 @@ def initialize(http_transport: Datadog::Core::Telemetry::Http::Transport.new)
         # Retrieves and emits a TelemetryRequest object based on the request type specified
         # @param request_type [String] the type of telemetry request to collect data for
         # @param data [Object] arbitrary object to be passed to the respective `request_type` handler
-        def request(request_type, data: nil)
+        def request(request_type, data: nil, payload: nil)
+          if payload && data
+            Datadog.logger.debug('Can not provide data and payload when generating temetry request')
```

Comment:
There's a typo in the message:

```suggestion
            Datadog.logger.debug('Cannot provide data and payload when generating telemetry request')
```

...but more than that, providing `data` and `payload` looks like a bug. So if it's a bug, have you considered raising an `ArgumentError` to make the error visible? That would make it more obvious when we're doing the wrong thing, whereas I can imagine the debug log here being lost under the radar.

Reply:
I thought about raising an exception. But I went against it because using an exception to control the flow felt weird. 

We already handle the exception when executing the HTTP operation, and if things fail, we log and return a value that, by the look of it, we never check, so I thought just adding some logging would be enough.

If you feel like we should through an exception I'm happy to change the code 😄

Reply:
I'm always a bit wary of "swallowing" errors -- in some cases we need to do it, but I'm not entirely convinced this is one of them. Do you have any concerns here with using an exception?

Reply:
> If we accidentally misuse the API, we should not break the customers' application.

Well... I'm not sure if I agree that this is the best way of doing that :D

Although, I'll clarify that I don't think changing this is a blocker for the rest of the PR -- to me it's more of a "in general I think we should avoid swallowing errors". :thinking: 

My reasoning is -- if we want to make absolutely sure we don't break customer's apps, I think the way to go isto have a begin/rescue that encompasses the whole API call, starting from the customer code. Otherwise we may always have some accidental exception that bubbles back up. So it's a bit weird to me to say "here we won't raise an exception because" instead of "here we always rescue because".